### ISSUE TITLE
Support subscribing to hook namespaces

### ISSUE THREAD

**Issue by thasmo** [getkirby/kirby#485]
It would be awesome if it's possible to subscribe to a hook namespace instead to subscribe just to the hook directly.

Something like this:
```php
kirby()->hook('panel.page.*', function($page) {
  // your hook code
});
```
So. it would also be possible to do things like these:
```php
kirby()->hook('panel.*', function($page) {
  // your hook code
});
// and
kirby()->hook('*', function($page) {
  // your hook code
});
```

To be able to know which event was triggered, the hook name could be passed as a second, optional parameter:
```php
kirby()->hook('*', function($page, $hook) {
  print $hook; // "panel.page.create"
});
```

I haven't seen any other hooks than those within the `panel` namespace, but in the future, when there're other namespaces, maybe, this could make sense also:
```php
kirby()->hook('*.page.*', function($page, $hook) {
  print $hook; // "api.page.delete"
});
```

**Comment by jenstornell** [comment]
Related:

https://forum.getkirby.com/t/what-hook-type-is-running/4976

**Comment by lukasbestle** [comment]
We were discussing this feature internally and unfortunately we can't implement it at this point to preserve backwards-compatibility.

Some hooks have more than one argument and new hooks in the future or hooks from plugins may even have more than that. So we can't add the hook name to the end of the argument list when using wildcards as the argument where the hook name appears would be a different one each time.

We also can't add it to the beginning of the list as plugins that implement hooks currently are expecting a different object there.

If you have an idea how to solve this, please tell us. If we find a good solution we might consider this feature.

**Comment by jenstornell** [comment]
Alright, no namespace at the moment. What about the other thing, adding the possibility to have the hook name as parameter? Would that still be something to implement (without namespace)?

**Comment by lukasbestle** [comment]
No, because the hook name as the parameter is the main reason why all doesn't work. Adding the param to the end is not clean.

But, surprise surprise: We might have actually found a good way to implement both. Stay tuned. :)

**Comment by lukasbestle** [comment]
This is now implemented and will be part of Kirby 2.4.

The magic is a new `Event` object that is available as `$this` inside the hook:

```php
kirby()->hook('panel.page.*', function($page, $oldPage = null) {
  var_dump($this->type()); // panel.page.update
});
```
